### PR TITLE
Story block: Handle blavatar_url for site icon

### PR DESCRIPTION
**Pull request by aforcier**
Fixes https://github.com/Automattic/story-block/issues/13.

#### Changes proposed in this Pull Request:
It seems that for some sites `get_site_icon_url` returns the fallback value, even when a custom site icon has been set. In those cases `blavatar_url` needs to be used. I'm not sure what the conditions for this are, I haven't been able to reproduce it on my own sites, but I have an example site where this occurs I can share.

I noticed that the `Jetpack_PostImages` class (and a few other places) has [logic for using `blavatar_` functions in addition to `get_site_icon_url`](https://github.com/Automattic/jetpack/blob/ba2dd6c30be137baad96aafb2466d0aed7e182e5/projects/plugins/jetpack/class.jetpack-post-images.php#L479-L494), so I added similar functionality to the story block. That seems to resolve the issue.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

* On a site with a custom site icon, play a story and make sure the site icon shows up at the top
* For a site that relies on blavatar (ping me for an example), play a story and make sure the site icon shows up

### CODE REVIEW

**Review by Tug — APPROVED** [review]
Not really confident about this but I believe blavatars were a jetpack feature which was deprecated once site icons became a feature in core. So that makes sense that we would need custom logic for this in jetpack.

The code looks sane to me and worked well.

:shipit:

File: projects/plugins/jetpack/extensions/blocks/story/story.php [hunk 1]
```diff
@@ -182,6 +182,30 @@ function get_image_crop_class( $width, $height ) {
...
 
+/**
+ * Returns a URL for the site icon.
+ *
+ * @param int    $size - Size for (square) sitei icon.
+ * @param string $fallback - Fallback URL to use if no site icon is found.
+ *
+ * @returns string
+ */
+function get_blavatar_or_site_icon_url( $size, $fallback ) {
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/story/story.php:193]: Would it be worth pulling this from the `Jetpack_PostImages` class instead?
https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/class.jetpack-post-images.php#L472-L477

Comment (aforcier) [reply projects/plugins/jetpack/extensions/blocks/story/story.php:194]: Sure! I refactored it to rely on `Jetpack_PostImages` in acbcc8 - let me know if you had a different approach than that in mind.

**Review by Tug — APPROVED** [review]
Still tests well for me

**Review by jeherve — APPROVED** [review]
This looks good to me. 🚢

**Comment by aforcier** [comment]
Deployed to WPCOM with changeset id 226972.

**Pull request merged**